lib/filters/nestedQueryFilter.js

```diff
@@ -0,0 +1,20 @@
+'use strict';
+
+/**
+ * @class  Elasticsearch Nested Query filter.
+ * @augments {Filter}
+ * @param {string} path   Nest path for document to filter.
+ * @param {Filter} filter Elastic filter instance to filter on the path.
+ */
+function NestedQueryFilter(path, filter) {
```

Comment:
@speddi To avoid confusion between queries and filters, can we change this to just NestedQuery?